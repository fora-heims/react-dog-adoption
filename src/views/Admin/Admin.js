import './Admin.css';

import React from 'react';
import Header from '../../components/Header/Header';
import Form from '../../components/Form/Form';
import { useState } from 'react';
import { insertDog } from '../../services/dogs';

export default function Admin() {
  const [dog, setDog] = useState({});
  const [message, setMessage] = useState('');

  const saveButtonHandler = async () => {
    try {
      await insertDog(dog);
      setMessage('Details successfully Updated!');
      setDog({});
      setTimeout(() => {
        setMessage('');
      }, 2000);
    } catch {
      setMessage('Something went wrong!');
    }
  };

  const updateDogState = (key, value) => {
    dog[key] = value;
    setDog({ ...dog });
  };

  return (
    <div className="admin">
      <Header />
      <Form
        message={message}
        saveButtonHandler={saveButtonHandler}
        {...dog}
        updateDogState={updateDogState}
      />
    </div>
  );
}
