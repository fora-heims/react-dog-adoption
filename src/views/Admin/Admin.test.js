import Admin from './Admin.js';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';

test('', async () => {
  const { container } = render(
    <MemoryRouter>
      <Admin />
    </MemoryRouter>
  );
  await screen.findByText('Enter Dog Details');
  expect(container).toMatchSnapshot();
});
