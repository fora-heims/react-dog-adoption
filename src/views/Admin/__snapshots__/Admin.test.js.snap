// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <div
    class="admin"
  >
    <header>
      <a
        aria-current="page"
        class="navigate activenavigate active"
        href="/"
      >
        Home
      </a>
      <a
        class="navigate activenot-active navigate"
        href="/admin"
      >
        Admin
      </a>
    </header>
    <div
      class="form"
    >
      <h2>
        Enter Dog Details
      </h2>
      <p />
      <label>
        Name:
        <input
          placeholder="Name"
          type="text"
          value=""
        />
      </label>
      <label>
        Bio:
        <textarea
          placeholder="Bio"
        />
      </label>
      <label>
        Age:
        <input
          placeholder="Age"
          type="number"
          value=""
        />
      </label>
      <label>
        Image:
        <input
          placeholder="Image URL"
          type="text"
          value=""
        />
      </label>
      <label>
        Breed:
        <input
          placeholder="Breed"
          type="text"
          value=""
        />
      </label>
      <button
        class="button"
      >
        Save
      </button>
    </div>
  </div>
</div>
`;
